splito/lohi/_utils.py

```diff
@@ -0,0 +1,42 @@
+from rdkit import Chem, DataStructs
+from rdkit.Chem import AllChem
+import numpy as np
+
+
+def get_similar_mols(lhs_smiles, rhs_smiles, return_idx=False):
+        """
+        Calculated maximal similarity between two sets for each molecule.
+
+        Parameters:
+            lhs_smiles -- list of smiles
+            rhs_smiles -- list of smiles
+            return_idx -- if True also returns idx of the similar molecules
+
+        Returns:
+            if return_idx = False:
+                nearest_sim -- list of length of lhs. i'th element contains maximal similarity between lhs[i] and rhs
+            if return_idx = True:
+                (nearest_sim, nearest_idx)
+                nearest_idx -- list of length of lhs. i'th element contains idx of rhs molecule, which is similar to lhs[i]
+        """
+        lhs_mols = []
+        for smiles in lhs_smiles:
+            lhs_mols.append(Chem.MolFromSmiles(smiles))
+        lhs_fps = [AllChem.GetMorganFingerprintAsBitVect(x, 2, 1024) for x in lhs_mols]
+
+        rhs_mols = []
+        for smiles in rhs_smiles:
+            rhs_mols.append(Chem.MolFromSmiles(smiles))
+        rhs_fps = [AllChem.GetMorganFingerprintAsBitVect(x, 2, 1024) for x in rhs_mols]
+
+        nearest_sim = []
+        nearest_idx = []
+        for lhs in lhs_fps:
+            sims = DataStructs.BulkTanimotoSimilarity(lhs, rhs_fps)
+            nearest_sim.append(max(sims))
+            nearest_idx.append(np.argmax(sims))
```

Comment:
You can use [Datamols `dm.cdist`](https://docs.datamol.io/stable/api/datamol.similarity.html#datamol.similarity.cdist) for this. That way, you can also leverage the built-in parallelization of `dm.cdist`.

Reply:
Changed the logic, so no need in this function.